### PR TITLE
[DO NOT MERGE YET] chore: V6 Getting Started - React, React Native, Angular, Vue & Vanilla JS

### PR DESCRIPTION
**Pull request by nadetastic**
EDIT: Updated to include React Native

#### Description of changes:
Updates for V6 in getting started for React, Angular, Vue, and Vanilla JS
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

File: src/fragments/cli-config.mdx [hunk 1]
```diff
@@ -1,10 +1,10 @@
-## Configure the Amplify CLI 
+## Configure the Amplify CLI
 
 To set up the Amplify CLI on your local machine, you have to configure it to connect to your AWS account.
 
 > If you already have an AWS profile with credentials on your machine, you can skip this step.
 
-> If you are using CLI version < 10.8.0 go to [Configure CLI < 10.8.0](#configure-the-amplify-cli-version--1080)
+> If you are using CLI version < 12 go to [Configure CLI < 10.8.0](#configure-the-amplify-cli-version--1080)
```
Comment (renebrandel) [inline src/fragments/cli-config.mdx:7]: Is this removed by error? The linked section seems to be deleted below.

File: src/fragments/start/getting-started/angular/data-model.mdx [hunk 2]
```diff
@@ -9,30 +9,30 @@ amplify add api
...
   id: ID!
   name: String!
   description: String!
   city: String!
-} 
+}
 ```
 
-The GraphQL Transform Library provides custom directives you can use in your schema that allow you to do things like define data models, set up authentication and authorization rules, configure serverless functions as resolvers, and more.  In the example schema above, the `@model` directive is part of Amplify's [GraphQL transformer](/cli/graphql/data-modeling) functionality and the `@auth` directive determines what [authorization rules](/cli/graphql/authorization-rules/) to apply. 
+The GraphQL Transform Library provides custom directives you can use in your schema that allow you to do things like define data models, set up authentication and authorization rules, configure serverless functions as resolvers, and more. In the example schema above, the `@model` directive is part of Amplify's [GraphQL transformer](/cli/graphql/data-modeling) functionality and the `@auth` directive determines what [authorization rules](/cli/graphql/authorization-rules/) to apply.
```
Comment (renebrandel) [inline src/fragments/start/getting-started/angular/data-model.mdx:35]:
```suggestion
Amplify's GraphQL API provides custom directives you can use in your schema that allow you to do things like define data models, set up authentication and authorization rules, configure serverless functions as resolvers, and more. In the example schema above, the `@model` directive is part of Amplify's [GraphQL API](/cli/graphql/data-modeling) functionality and the `@auth` directive determines what [authorization rules](/cli/graphql/authorization-rules/) to apply.
```

File: src/fragments/start/getting-started/angular/data-model.mdx [hunk 3]
```diff
@@ -87,32 +87,24 @@ Open `src/main.ts` and add the following code to configure the Angular project w
 
 ```typescript
 import { Amplify } from 'aws-amplify';
-import aws_exports from './aws-exports';
-Amplify.configure(aws_exports);
+import amplifyconfig from './amplifyconfiguration.json';
+Amplify.configure(amplifyconfig);
```
Comment (renebrandel) [inline src/fragments/start/getting-started/angular/data-model.mdx:91]:
```suggestion
import amplifyconfig from './amplifyconfiguration.json';

Amplify.configure(amplifyconfig);
```

**Review by renebrandel — CHANGES_REQUESTED** [review]
Found issues in PR, going to push changes soon. Primarily subscriptions related.

File: src/fragments/start/getting-started/reactnative/auth.mdx [hunk 4]
```diff
@@ -168,6 +168,8 @@ const SignOutButton = () => {
 const initialFormState = {name: '', description: ''};
 
 const App = () => {
+
+  const client = generateClient();
   const [formState, setFormState] = useState(initialFormState);
```
Comment (renebrandel) [inline src/fragments/start/getting-started/reactnative/auth.mdx:173]:
```suggestion
const client = generateClient();

const App = () => {
  const [formState, setFormState] = useState(initialFormState);
```

File: src/fragments/start/getting-started/reactnative/auth.mdx [hunk 5]
```diff
@@ -285,6 +291,9 @@ const SignOutButton = () => {
 const initialFormState = {name: '', description: ''};
 
 const App = () => {
+
+  const client = generateClient();
+
```
Comment (renebrandel) [inline src/fragments/start/getting-started/reactnative/auth.mdx:296]:
```suggestion
const client = generateClient();
  
const App = () => {
```

File: src/fragments/start/getting-started/reactnative/data-model.mdx [hunk 6]
```diff
@@ -158,17 +163,20 @@ import {
...
-Amplify.configure(awsExports);
+import amplifyconfig from './src/amplifyconfiguration.json';
+Amplify.configure(amplifyconfig);
 
 const initialState = {name: '', description: ''};
 
 const App = () => {
+
+  const client = generateClient();
+
```
Comment (renebrandel) [inline src/fragments/start/getting-started/reactnative/data-model.mdx:179]:
```suggestion
const client = generateClient();

const App = () => {
```

**Comment by nadetastic** [comment]
Changes merged Via Different PR - https://github.com/aws-amplify/docs/pull/6305

**Pull request closed without merging**